src/software/world/ball.h

```diff
@@ -61,13 +66,29 @@ class Ball final
      */
     Timestamp lastUpdateTimestamp() const;
 
+    /**
+     * Gets the previous timestamps for each state stored in states_
+     *
+     * @return Vector containing the update timestamp history starting with the oldest
+     * available data at index 0
```

Comment:
Isn't the newest data at the "front" of the buffer which is index 0?

Reply:
The old class was using push_back so I used it and I think it loops from front to back so it'll be the oldest one

Reply:
Ah I see. This seems inconsistent with the order we store states internally (newest states at the front). My expectation when calling a function like this would be it would go newest -> oldest. I think this would be a more common use case since the caller may only want a subset of the history, but that would generally always start at the newest data (ie. a caller usually won't ask for history in the range [t-1, t-3], it would be [t, t-2] if that makes sense).

@garethellis0 @matthewberends @jsn0716 thoughts on convention for this?

Reply:
I think either are okay as long as it's consistent. What data structure / operations are used for storing states internally?

Reply:
A circular buffer with a fixed size, so that old items get popped automatically when new ones are added if the buffer is at capacity.

Reply:
I see, yeah I think it would be nice to have this consistent, but maybe in a separate PR

Reply:
Oh fair point since anything using this is expecting the older interface. Made #995